src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -48,6 +51,26 @@ export class WorkspaceIdConsumerWrapper {
     return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
   }
 
+  // If the object.workspaces is null/[] (as global object), return it directly.
+  // If the workspace is specified, validate whether the object exists in the workspace.
+  private validateObjectExistInWorkspaces<T>(
+    object: SavedObject<T>,
+    workspace: string
+  ): SavedObject<T> {
+    if (object.workspaces && object.workspaces.length > 0) {
+      if (!object.workspaces.includes(workspace)) {
```

Comment:
general question, and im not really sure if this could be possible because I don't think workspaces references any associated saved objects

if the `object.workspaces` does not include the current workspace is there ever a `workspace.objects` or something like that? or only the object keeps reference to the workspace?

Reply:
Hi @kavilla , only the object keeps reference to the workspace like `object.workspaces` and there is no `workspace.objects`